plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/extractor/MetadataCollectorTest.java

```diff
@@ -164,4 +164,226 @@ void testPlugin() throws Exception {
         // Absent
         assertFalse(pluginMetadata.hasFile(ArchetypeCommonFile.WORKFLOW_CD));
     }
+
+    @Test
+    void testPluginWithJenkinsfile() {
+        rewriteRun(
+                // language=groovy
+                groovy(
+                        """
+                         buildPlugin(
+                         useContainerAgent: true,
+                         configurations: [
+                                [platform: 'linux', jdk: 21],
+                                [platform: 'windows', jdk: 21],
+                         ])
+                         """,
+                        spec -> spec.path("Jenkinsfile")),
+                // language=xml
+                pomXml(
```

Comment:
This test only check Jenkinsfile. Let's try to use a minimal pom. And if possible from a constant. Will make the test more easy to read